smiDE post

dan11mcguire opened the pull request:
A very simple scratch space post with links to DE preprint + NanoStringBiostats/smiDE github repo (currently private).

Main vignette for the package in the README of github page.

dan11mcguire committed:
smiDE post

dan11mcguire committed:
update reference and text

eveilyeverafter reviewed (changes requested):
It looks okay as-is but it can be enhanced by:

- adding a statement that the package vignette is there for further exploration.
- It would be great if you could include a figure for the main blog page as well.

Right now the smiDE repo is not accessible so the links do not work. Having that repo be public is gating this post.

dan11mcguire commented:
> It looks okay as-is but it can be enhanced by:
> 
>     * adding a statement that the package vignette is there for further exploration.
> 
>     * It would be great if you could include a figure for the main blog page as well.
> 
> 
> Right now the smiDE repo is not accessible so the links do not work. Having that repo be public is gating this post.

i will jazz it up

dan11mcguire committed:
beefed up

dan11mcguire committed:
figures for jazzed up blog post

dan11mcguire commented:
@eveilyeverafter , I've jazzed up the post with  detailed walk through of main sections in the paper and added a bunch of figures.
Ready for your review again when you have time.

patrickjdanaher commented:
Comments on the post itself:
- Avoid words like "exciting", which lacks oomph. 
- 2nd figure, examples of DE questions: spell it out a little more in the bullet points, for example, "How does the expression of genes in T cells differ when located in stroma vs. tumor bed? (Left panel, blue vs. purple points.)
- "tumor-native epithelial cells" - I'd just say "cancer cells"
- "CD79A retains the significance in the SRE model" - delete "the"
- intro paragraph: I'd emphasize the value proposition here: legacy approaches will very often lead you astray, and smiDE is a solution.

patrickjdanaher commented:
REALLY nice package vignette. Only comment would be to provide a very visible commentary on runtime considerations and recommended models.

dan11mcguire committed:
address patricks comments; new opening paragraph + remaining bullet points

dan11mcguire commented:
@patrickjdanaher thanks, I think I've addressed your comments above for the post.  

Here is a new version for the intro paragraph, let me know any suggestions or edits

> Differential Expression Analysis With Spatially Correlated Data
> 
> Led by the brilliant Ana Gabriela Vasconcelos, we’re excited to present a preprint [“Differential Expression Analysis for Spatially Correlated Data”](https://www.biorxiv.org/content/10.1101/2024.08.02.606405v1.full) ([Vasconcelos et al. 2024](http://10.19.26.230/p/0748a441/posts/smiDE/#ref-Vasconcelos2024)), discussing opportunities and unique challenges fundamental to differential expression (DE) analyses with spatial data.
> 
> We find that traditional DE methods applied to spatial data result in high rates of false discovery, and benchmark performance of different modeling strategies in terms of false discovery rates, power, effect size estimation, and computational complexity.
> 
> We provide general methodology recommendations for how a researcher should apply DE in their own study to avoid potential pitfalls that can damage the integrity of an analysis. Tools for methods discussed in the paper are implemented in the R package [smiDE](https://github.com/Nanostring-Biostats/smiDE), with vignettes and examples on the package’s github page.
> 
> A brief summary of highlights and takeaways from the paper are described throughout the rest of this post.

dan11mcguire commented:
@patrickjdanaher fyi:  added commentary on runtime considerations and recommended models in the github README / vignette

dan11mcguire committed:
take evelyns suggestion with small tweak

dan11mcguire committed:
take evelyns suggestion with small tweak

dan11mcguire committed:
use srt for spatially-resolved transcriptomics

dan11mcguire committed:
update the date

dan11mcguire committed:
change language for overlapping cells

dan11mcguire committed:
test

dan11mcguire committed:
add a comment

dan11mcguire committed:
add smiDE as a submodule

dan11mcguire commented:
@eveilyeverafter @patrickjdanaher 

I've added `smiDE` package as a submodule in this PR branch.
I've made INLA package a suggestion, and package will prompt user to install INLA if that method is specified, but otherwise works for other methods.
```
"'GP_INLA' is specified but INLA package is not installed.  To run a spatial model using INLA 'GP_INLA', please install the INLA R package: https://www.r-inla.org/download-install"
```

I've also updated the language for all of the functions, per Prajan's requirement.
There are no mentions of "contamination".  Instead we now have i.e., "overlap_cells_ratio_metric" instead of "contamination_ratio_metric".

Post and README are also updated now to reflect the new language.

eveilyeverafter commented:
Is the plan now to have the smiDE package within this repo instead of having it be a standalone repo?

dan11mcguire committed:
merge in latest from Main

dan11mcguire committed:
cleanup

dan11mcguire committed:
update link to package

dan11mcguire committed:
Merge commit '5fe4b493200368211c71a1db0f435796c7359d5a' as '_code/smiDE'

dan11mcguire committed:
Squashed '_code/smiDE/' content from commit 354d11b

git-subtree-dir: _code/smiDE
git-subtree-split: 354d11b10dcefa384ff9782a87bbd31ffe54164d

dan11mcguire committed:
fix the link
